src/main/java/io/quarkus/gizmo2/creator/ops/StringBuilderOps.java

```diff
@@ -0,0 +1,82 @@
+import io.quarkus.gizmo2.Expr;
+import io.quarkus.gizmo2.creator.BlockCreator;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Operations on {@link StringBuilder}.
+ */
+public final class StringBuilderOps extends ObjectOps implements ComparableOps {
+
+    /**
+     * Construct a new instance on an existing {@link StringBuilder}.
+     *
+     * @param bc the block creator (must not be {@code null})
+     * @param obj the receiver string builder (must not be {@code null})
+     */
+    public StringBuilderOps(final BlockCreator bc, final Expr obj) {
+        super(StringBuilder.class, bc, obj);
+    }
+
+    /**
+     * Construct a new instance on a newly created {@link StringBuilder}.
+     * The {@code StringBuilder} instance may be obtained using {@link #getInstance()}.
+     *
+     * @param bc the block creator (must not be {@code null})
+     */
+    public StringBuilderOps(final BlockCreator bc) {
+        super(StringBuilder.class, bc, bc.define("stringBuilder", bc.new_(StringBuilder.class)));
+    }
+
+    /**
+     * Construct a new instance on a newly created {@link StringBuilder}
+     * with given {@code capacity}. The {@code StringBuilder} instance
+     * may be obtained using {@link #getInstance()}.
+     *
+     * @param bc the block creator (must not be {@code null})
+     * @param capacity the capacity of the newly created {@link StringBuilder}
+     */
+    public StringBuilderOps(final BlockCreator bc, final int capacity) {
+        super(StringBuilder.class, bc, bc.new_(StringBuilder.class, Constant.of(capacity)));
+    }
+
+    public StringBuilderOps append(final Expr expr) {
+        switch (expr.type().descriptorString()) {
+            case "Z" -> invokeInstance(StringBuilder.class, "append", boolean.class, expr);
+            case "B", "S", "I" -> invokeInstance(StringBuilder.class, "append", int.class, expr);
+            case "J" -> invokeInstance(StringBuilder.class, "append", long.class, expr);
+            case "F" -> invokeInstance(StringBuilder.class, "append", float.class, expr);
+            case "D" -> invokeInstance(StringBuilder.class, "append", double.class, expr);
+            case "C" -> invokeInstance(StringBuilder.class, "append", char.class, expr);
+            case "[C" -> invokeInstance(StringBuilder.class, "append", char[].class, expr);
+            case "Ljava/lang/String;" -> invokeInstance(StringBuilder.class, "append", String.class, expr);
+            case "Ljava/lang/CharSequence;" -> invokeInstance(StringBuilder.class, "append", CharSequence.class, expr);
+            default -> invokeInstance(StringBuilder.class, "append", Object.class, expr);
+        }
+        return this;
+    }
+
+    public StringBuilderOps append(final char constant) {
+        return append(Constant.of(constant));
+    }
+
+    public StringBuilderOps append(final String constant) {
+        return append(Constant.of(constant));
+    }
+
+    public Expr callToString() {
```

Comment:
There's already `io.quarkus.gizmo2.creator.ops.ObjectOps.objToString()`...

Reply:
Ah, good point! Indeed this can be reused. Will fix.

Reply:
Done.